pkgs/build-support/ocaml/dune.nix

```diff
@@ -7,8 +7,11 @@ let Dune =
   { "1" = dune_1; "2" = dune_2; "3" = dune_3; }."${dune-version}"
 ; in
 
-if (args ? minimumOCamlVersion && lib.versionOlder ocaml.version args.minimumOCamlVersion) ||
-   (args ? minimalOCamlVersion && lib.versionOlder ocaml.version args.minimalOCamlVersion)
+if (args ? minimumOCamlVersion)
+then throw "${pname}-${version}: minimumOCamlVersion is deprecated and was replaced with minimalOCamlVersion"
```

Comment:
What’s the purpose of this error message? When should this peace of code be removed from `nixpkgs`?

Reply:
For whoever is still using `minimumOCamlVersion` in their projects, this warns and suggests to replace it with `minimalOCamlVersion`.

Do you want to remove this?

Reply:
Not sure. It feels like you are reintroducing technical dept when you wanted to remove it instead.

Also, if you just dropped this line, you could let `ocaml-syntax-shims` as is and avoid the unnecessary mass-rebuild.

Reply:
Drop changes to file "pkgs/build-support/ocaml/dune.nix" then?

Drop changes to file "ocaml-syntax-shims" as well?

Reply:
Dropped changes to:
- pkgs/build-support/ocaml/dune.nix
- ocaml-syntax-shims

Reply:
Well, you can change `buildDunePackage`: I think it is a good idea to remove the functionality attached to the argument `minimumOCamlVersion`.